src/libs/import.c

```diff
@@ -694,8 +703,9 @@ static void _lib_import_update_preview(GtkFileChooser *file_chooser, gpointer da
       g_object_unref(loader); // This should clean up tmp as well
     }
   }
-  if(have_preview && !no_preview_fallback)
+  if(!have_preview && !no_preview_fallback)
```

Comment:
Nope, that is the wrong logic. The rotation has to be done when getting the thumbnail. You disabled it for when the first attempt worked out.
Calling `gdk_pixbuf_new_from_file_at_size()` has to happen iff the previous block failed, and the rotation logic has to be independent from that. It has to be determined if the rotation logic needs to be used when opening the whole file, too.

Reply:
OK, now I get it. 
Still not clear though the case of DNG files: if we try to open a DNG file, `no_preview_fallback` will be TRUE. 
* it OK to read the DNG file with `gdk_pixbuf_new_from_file_at_size()` and **only** the thumbnail reading should be prevented, or
* both the thumbnail reading **and**  `gdk_pixbuf_new_from_file_at_size()` call should be blocked for those files?

Reply:
Looking at the git log it seems that passing a DNG to `gdk_pixbuf_new_from_file_at_size()` is error prone: a56251caa00f4d73056672db42b441ca043d47e9

Reply:
OK, the latest commit does not call `gdk_pixbuf_new_from_file_at_size()` for DNG files.
As this comment is from 5 years ago, it might be that since then this has been fixed in libtiff(?) - but I cannot test it as I don't have hdr DNG file at hand.